dataplug/app/com/hubofallthings/dataplug/controllers/Application.scala

```diff
@@ -175,9 +179,21 @@ class Application @Inject() (
       .recover {
         case e =>
           logger.error(s"$provider API cannot be accessed: ${e.getMessage}. Redirecting to $provider OAuth service.", e)
-          Redirect(com.hubofallthings.dataplug.controllers.routes.SocialAuthController.authenticate(provider))
+          Redirect(dataPlugViewSet.disconnectRedirect)
       }
   }
 
+  def disconnect(): Action[AnyContent] = SecuredAction.async { implicit request =>
+    val userId = request.identity.linkedUsers.head.userId
```

Comment:
Can `.head` explode here?  Consider `.headOption`

Reply:
Yes it can. That's totally my bad, not intended. I definitely will fix this

Reply:
👍

Reply:
@whiteshadow-gr not fixed yet?

Reply:
@augustinas no changes made. Was waiting for your review and go and fix everything mentioned in one go. 👍